TemplateTagsTestCase.test_simple fails in my environment

### Describe the issue

Hi, thank you for maintaining this software and keeping it in such a good shape! I am surprised to repeatedly encounter a failing test. Since 5.14 `TemplateTagsTestCase.test_simple` fails, this persists even in 5.14.3 and I am unable to find the cause. Could you please give me some advice on what could be the cause? Thanks a lot!

I don't upper-bound the deps, so maybe something got out of the bounds, but I checked the deps I thought could be related and all were in.

### I already tried

- [x] I've read and searched [the documentation](https://docs.weblate.org/).
- [x] I've searched for similar filed issues in this repository.

### Steps to reproduce the behavior

Run tests via plain pytest call

### Expected behavior

tests passing

### Screenshots

_No response_

### Exception traceback

```pytb
[ 3913s] =================================== FAILURES ===================================
[ 3913s] _______________________ TemplateTagsTestCase.test_simple _______________________
[ 3913s] 
[ 3913s] self = <weblate.accounts.tests.test_templatetags.TemplateTagsTestCase testMethod=test_simple>
[ 3913s] 
[ 3913s]     def test_simple(self) -> None:
[ 3913s]         template = Template("""
[ 3913s]                 {% load site_url %}
[ 3913s]                 <html><body>
[ 3913s]                 {% filter add_site_url %}
[ 3913s]                     text:
[ 3913s]                     <a href="/foo"><span>Foo</span></a>
[ 3913s]     
[ 3913s]                 {% endfilter %}
[ 3913s]                 </body>
[ 3913s]                 </html>
[ 3913s]             """)
[ 3913s] >       self.assertHTMLEqual(
[ 3913s]             """
[ 3913s]             <html>
[ 3913s]             <body>
[ 3913s]                 <p>
[ 3913s]                     text:
[ 3913s]                     <a href="http://example.com/foo">
[ 3913s]                         <span>
[ 3913s]                             Foo
[ 3913s]                         </span>
[ 3913s]                     </a>
[ 3913s]                 </p>
[ 3913s]             </body>
[ 3913s]             </html>
[ 3913s]             """,
[ 3913s]             template.render(Context()),
[ 3913s]         )
[ 3913s] E       AssertionError: <html>
[ 3913s] E       <body>
[ 3913s] E       <p>
[ 3913s] E       text:<a href="http://example.com/foo">
[ 3913s] E       <span>
[ 3913s] E       Foo
[ 3913s] E       </span>
[ 3913s] E       </a> [truncated]... != <html>
[ 3913s] E       <body>
[ 3913s] E       <a href="http://example.com/foo">
[ 3913s] E       <span>
[ 3913s] E       Foo
[ 3913s] E       </span>
[ 3913s] E       </a>
[ 3913s] E       </body>
[ 3913s] E        [truncated]...
[ 3913s] E         <html>
[ 3913s] E         <body>
[ 3913s] E       - <p>
[ 3913s] E       - text:<a href="http://example.com/foo">
[ 3913s] E       ? -----
[ 3913s] E       
[ 3913s] E       + <a href="http://example.com/foo">
[ 3913s] E         <span>
[ 3913s] E         Foo
[ 3913s] E         </span>
[ 3913s] E         </a>
[ 3913s] E       - </p>
[ 3913s] E         </body>
[ 3913s] E         </html>
```

### How do you run Weblate?

Other

### Weblate versions

5.14 -- 5.14.3

### Weblate deploy checks

```shell

```

### Additional context

https://build.opensuse.org/package/show/home:mcalabkova:branches:devel:languages:python/weblate, you know the environment

--- a/weblate/accounts/templatetags/site_url.py
+++ b/weblate/accounts/templatetags/site_url.py
@@ -5,12 +5,17 @@
 """Convert any links in HTML to absolute."""
 
 from io import StringIO
+from typing import TYPE_CHECKING
 
 from django import template
+from django.utils.html import escape
 from django.utils.safestring import mark_safe
 from lxml import etree
 
 from weblate.utils.site import get_site_url
+
+if TYPE_CHECKING:
+    from django.utils.safestring import SafeString
 
 register = template.Library()
 
@@ -19,24 +24,26 @@ register = template.Library()
 def add_site_url(content):
     """Automatically add site URL to any relative links or images."""
     parser = etree.HTMLParser(collect_ids=False)
-    tree = etree.parse(StringIO(content), parser)
+    tree = etree.parse(StringIO(f"<html><body>{content}</body></html>"), parser)
     for link in tree.findall(".//a"):
         url = link.get("href")
-        if url.startswith("/"):
+        if url and url.startswith("/"):
             link.set("href", get_site_url(url))
     for link in tree.findall(".//img"):
         url = link.get("src")
-        if url.startswith("/"):
+        if url and url.startswith("/"):
             link.set("src", get_site_url(url))
 
     body = tree.find("body")
+    if body is None:
+        msg = "Failed parsing HTML!"
+        raise ValueError(msg)
+    parts: list[str | SafeString] = [
+        etree.tostring(child, pretty_print=True, method="html", encoding="unicode")
+        for child in body.iterchildren()
+    ]
+    if body.text:
+        parts.insert(0, escape(body.text))
     return mark_safe(  # noqa: S308
-        "".join(
-            [
-                etree.tostring(
-                    child, pretty_print=True, method="html", encoding="unicode"
-                )
-                for child in body.iterchildren()
-            ]
-        )
+        "".join(parts)
     )

--- a/weblate/accounts/tests/test_templatetags.py
+++ b/weblate/accounts/tests/test_templatetags.py
@@ -9,15 +9,21 @@ from weblate.accounts.templatetags.urlformat import urlformat
 
 
 class TemplateTagsTestCase(SimpleTestCase):
-    def test_simple(self) -> None:
+    def test_add_site_url_filter(self) -> None:
         template = Template("""
                 {% load site_url %}
                 <html><body>
                 {% filter add_site_url %}
+                <p>
                     text:
                     <a href="/foo"><span>Foo</span></a>
-
+                </p>
                 {% endfilter %}
+                <p>
+                {% filter add_site_url %}
+                    other&amp;
+                {% endfilter %}
+                </p>
                 </body>
                 </html>
             """)
@@ -33,6 +39,7 @@ class TemplateTagsTestCase(SimpleTestCase):
                         </span>
                     </a>
                 </p>
+                <p>other&amp;</p>
             </body>
             </html>
             """,

